File: NicePartUsage.Domain/Models/NpuCreation.cs
using System.ComponentModel.DataAnnotations;

namespace NicePartUsage.Domain.Models
{
    public class NpuCreation
    {
        [Key]
        public int Id { get; set; }
        public List<Element> Elements { get; set; }

        [Required]
        [MaxLength(100)]
        public string Title { get; set; }

        [MaxLength(500)]
        public string Description { get; set; }

        [Required]
        public byte[] FileData { get; set; }

        public DateTime UploadDate { get; set; } = DateTime.UtcNow;

        public virtual User Creator { get; set; }

        public virtual List<Score> Scores { get; set; }

        public virtual ICollection<NpuCreationElement> NpuCreationElements { get; set; }
    }
}

File: nicepartusage.client/src/models/NpuCreation.ts
import { Element, NpuCreationElement, Score, User } from '.';

export interface NpuCreation {
    id: number;
    elements: Element[];
    title: string;
    description: string;
    fileData: string;
    uploadDate: string;
    creator: User;
    scores: Score[];
    npuCreationElements: NpuCreationElement[];
}